DOP-4671 - detect system for light/dark modes

seungpark opened the pull request:
### Stories/Links:

DOP-4671

### Current Behavior:

_Put a link to current staging or production behavior, if applicable_

### Staging Links:

[staged site that should read from your system settings](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/drivers/seung.park/DOP-4671/index.html)

### Notes:
- Local testing requires env variable `GATSBY_ENABLE_DARK_MODE=true` to have all the effects take place (from [this PR](https://github.com/mongodb/snooty/pull/1099))
- To test system detection, change your system appearance to light/dark and staged site should respect your system settings
- Added to the previous PR's logic of saving to the local storage - the HTML document will also be added a classname for `dark-theme` (or `light-theme`) via a pre render script that reads the local storage. This allows for our own styling ie. page background before hydration occurs. How this can be pre-rendered remains a mystery as we are serving static sites.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

seungpark commented on gatsby-ssr.js:
```diff
@@ -6,7 +6,7 @@ import { theme } from './src/theme/docsTheme';
 
 export const onRenderBody = ({ setHeadComponents }) => {
-  setHeadComponents([
+  const headComponents = [
```
this has already been reviewed in https://github.com/mongodb/snooty/pull/1099/files

seungpark commented on src/context/dark-mode-context.js:
```diff
@@ -8,45 +8,54 @@
-        >
-          Click to toggle Dark Mode
-        </Button>
+        {process.env['GATSBY_ENABLE_DARK_MODE'] === 'true' && (
```
from [previous PR](https://github.com/mongodb/snooty/pull/1099) to test darkmode

caesarbell reviewed (commented):
The new code change i.e. adding the dark mode toggle button and it's logic looks good, but the script tag from the previous PR, doesn't seem to work for me. I changed the appearance in the chrome browser to light, and the dark-theme class rendered, I only see the light-theme class when I click the toggle button.

seungpark commented:
> The new code change i.e. adding the dark mode toggle button and it's logic looks good, but the script tag from the previous PR, doesn't seem to work for me. I changed the appearance in the chrome browser to light, and the dark-theme class rendered, I only see the light-theme class when I click the toggle button.

ah the script was not running due to a syntax error. addressed this in [this commit](https://github.com/mongodb/snooty/pull/1098/commits/592dfb64ae9f90ff019a81caab6b24a69bf945d0). updated the staging link and it should be detecting *system theme*, not so much the chrome theme (ie. in mac, i updated my appearance to be dark). i will look into if we can detect a browser theme as well 👍

caesarbell commented on gatsby-ssr.js:
```diff
@@ -58,7 +58,7 @@ export const onRenderBody = ({ setHeadComponents }) => {
         dangerouslySetInnerHTML={{
           __html: `
-            function () {
+            !function () {
```
Good catch.

caesarbell reviewed (approved):
LGTM
